### PR TITLE
Solution HW 5.1

### PR DESCRIPTION
**Pull request by Bogdan-Demianchuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+    private Engine nameEngine;
+
+    @Override
+    public Car clone() {
+        Car clonedCar = new Car();
+        clonedCar.mark = this.mark;
+        clonedCar.type = this.type;
+        clonedCar.maxSpeed = this.maxSpeed;
+        clonedCar.color = this.color;
+        clonedCar.nameEngine = this.nameEngine;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:17]: Do we need this keywords here?How about create a constructor that would get all fields and use it?

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
-public class Engine {
+public class Engine implements Cloneable {
+    private int capacity;
+    private String typeFuel;
+    private boolean turbine;
+    private double cylinderBore;
+    private double torque;
 
+    @Override
+    public Engine clone() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:11]: same

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/Car.java]: Done

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/Engine.java:21]: Done

**Pull request closed without merging**